### PR TITLE
Add the ability to whitelist based on permissions and support push-anonymous permissions

### PR DESCRIPTION
**Pull request by stevielb**
This change supports filtering permissions in the connector before pushing to a target source via the permission_whitelist node. It also supports a config flag (which must be implemented by the implementation) for supporting pushing all videos as public to the target. The adds the implementation for Coveo, though other implementations would need to implement this on their own.

The cleanest design was to update the convert_to_target method to take a ConnectorConfig rather than it's element FieldMapping. This is in line with the other two functions needed for each implementation to be defined and should overall simplify the paradigm too.

The readme was updated with helpful examples, including how to use the whitelist to push all videos which are "All User" or "Public", as well as videos on a given user or group.

image

Tested via pushing content as "allowAnonymous" to a new coveo target, as well as verified that existing content pushed correctly to the original princpal secured push source. Used the debug implementation to verify all items. Tested the whitelist with a given user, a given group, All Users, and Public. Verified that the program will terminate with a helpful log if an incorrect whitelist element is given.

Additionally, tox succeeds and the standalone build work as well. Performed all testing using the standalone build artifact. I will cut a release after merge with the new "out of the box" connector.

### CODE REVIEW

**Review by hohno-panopto — APPROVED** [review]
Looks good to me.

File: README.md [hunk 1]
```diff
@@ -153,6 +153,26 @@ panopto_oauth_credentials:
...
 Running the debug connector once you have configured your user and your oauth credentials is recommended to confirm that you are set up to correctly talk to the Panopto APIs, before continuing to your final implementation.
 
+Other configuration options which are helpful to know about:
+```yml
+# Allows you to only send videos to the target matching one of a given permission set
+# First bullet would only sync videos with view permission of a given Panopto groups
+# The second for a given group from an external Identity provider
+# The third would be only videos with panopto public permissions
+# The fourth would be all authenticated users at your organization
+permission_whitelist:
```
Comment (kbaum-panopto) [inline README.md]: I could be missing something but should this be `principal_whitelist` instead?

**Review by kbaum-panopto — APPROVED** [review]
Left one small comment on the readme, but it's possible I just missed converting a public name to an internal name somewhere.

File: README.md (hunk 1 above)
Comment (knorman-panopto) [inline README.md]: Nit: We have started replacing all instances of whitelist with allowlist

**Review by knorman-panopto — COMMENTED** [review]
Just a small comment on the fact that we replaced whitelist with allowlist where we can. I don't know if this is part of the specific spec for what you are making so don't know if possible to change.

93770 and 93771 were the TFS items tracking that replacement.

**Pull request merged**